### ISSUE TITLE
https.request() do no get called when Content-length specified in options.

### ISSUE THREAD

**Issue by sanjaiganesh** [olalonde/follow-redirects#40]
I would like to know what is wrong in this code.. I don't https.Request() gets called when I remove content-length. 

 setProperty: function(name, value) {
        var deferred = q.defer();   // Take a deferral

```
    var postData = JSON.stringify({name : value});
    var options = {
        protocol: 'https:',
        host: 'developer-api.nest.com',
        path: '/devices/thermostats/' + deviceId,
        headers: {
            'Authorization': 'Bearer ' + accessToken,
            'Content-Type': 'application/json',
            'Accept': 'application/json',
            'Content-length': Buffer.byteLength(postData),
        },
        method: 'PUT',
    };

    var httpsRequest = https.request(options, (getRes) => {
        console.log("Request sent");
        var body = '';
        getRes.setEncoding('utf8');
        getRes.on('data', (data) => {
            body += data;
        });

        getRes.on('end', () => {
            if (getRes.statusCode != 200) {
                deferred.reject("Error Code:" + getRes.statusCode);
            } else {
                var data = JSON.parse(body);
                deferred.resolve(target_temperature_c);
            }
        });

        getRes.on('error', (e) => {
            deferred.reject(e);
        });
    });
    httpsRequest.write(postData);
    httpsRequest.end();

    return deferred.promise; // return the promise
},
```

**Comment by RubenVerborgh** [comment]
Are you sure your content length is correct? Because I expect `postData` to be encoded as `utf8` (since it's JSON), but you don't set this encoding, nor do you pass the encoding to `Buffer.byteLength`.

Does this work with the regular Node.js `https` module?

**Comment by mogaal** [comment]
I am having the same issue, if I commented Content-length (and pointing to directly to the destination URL instead of the redirect URL) and it works. I am passing header in the following way: Content-length': Buffer.byteLength(PostData, 'utf8'), I also try without the utf8 option but not luck :-(.

Any idea?

**Comment by RubenVerborgh** [comment]
@mogaal Could you make a minimal  working example, for instance with `http://scooterlabs.com/echo`?

**Comment by RubenVerborgh** [comment]
The test I added seems to indicate this is fixed, together with the solution for #35.
